### DISCUSSION TITLE
How to put files to subdirectory using SpatieMediaLibraryFileUpload?

### DISCUSSION THREAD

**Discussion by ig0r74** [filamentphp/filament#2073] [Help]
The option `->directory('one/two')` doesn't works.

<img width="287" alt="image" src="https://user-images.githubusercontent.com/6386398/161246945-d336f123-0c94-4b76-bf02-1339d6e2d2b0.png">

Thanks!

**Comment by danharrin** [comment]
What's the method for controlling the directory of the uploaded file using media library?

**Reply by ig0r74** [reply]
Hi, Dan! I don't know, I'm new to this package.

```
Filament\Forms\Components\SpatieMediaLibraryFileUpload::make('image')
    ->collection('brands')
    ->directory('shop/brands')
    ->image(),
```

**Reply by danharrin** [reply]
I mean that in Spatie's package, not ours, how do you set the directory when you add a file to media library? I've not done it before.

**Reply by ig0r74** [reply]
I searched a lot, and i think its only possible by setting media disk:
https://spatie.be/docs/laravel-medialibrary/v10/installation-setup#a-idcontent-adding-a-media-disk-1-hrefcontent-adding-a-media-disk-1-classanchor-link-aria-hiddentrue-titlepermalinkaa-idcontent-adding-a-media-disk-hrefcontent-adding-a-media-disk-classanchor-link-aria-hiddentrue-titlepermalinkaadding-a-media-disk

**Reply by danharrin** [reply]
Well, why are you asking why it doesn't work in Filament then? 😁
